### ISSUE TITLE
[Accessibility]: History panel view: read out more user-friendly list summary on open

### ISSUE THREAD

**Issue by lrhacker** [sourcegraph/sourcegraph#44589]
### Audit type

Keyboard navigation

### Problem description

Similar to #44583, this is an improvement on the existing accessibility work done in #44452.

For this, per [this comment](https://github.com/sourcegraph/sourcegraph/pull/44452#pullrequestreview-1185135622) from @courier-new:

> [W]hen it starts reading the history, there's no way of knowing how long it will take to finish unless you see the whole list yourself. Could we consider having a `<VisuallyHidden />` message with `aria-live="polite"` at the top of the ConnectionContainer that announces how many history items there are, or something similar, instead?

### Expected behavior

Read out a more user-friendly summary message (or similar) when the history panel opens. If the work to manage the focus from #44583 has already been done, there might be other options, as well (e.g. read out a list summary, then focus and read out the first item to start with). It might be worth coordinating with design on the message that's read aloud.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
